core/src/main/java/io/grpc/ServiceDescriptor.java

```diff
@@ -108,4 +132,89 @@ static void validateMethodNames(String serviceName, Collection<MethodDescriptor<
           "duplicate name %s", method.getFullMethodName());
     }
   }
+
+  /**
+   * Creates a new builder for a {@link ServiceDescriptor}.
+   *
+   * @since 1.1.0
+   */
+  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/2666")
+  public static Builder newBuilder() {
```

Comment:
I have come from writing a lot of proto code, where `build()` is when invariants are enforced.  I feel there are a lot of people who also have this expectations, just based on the number of them who write code at Google.  

@zhangkun83  thoughts?